File: dyswlongsubstr.cpp
#include<iostream>
#include<set>
using namespace std;

int lonSubStr(string baseStr){
 int left = 0;
 int right = 0;
 set<int> hashset;
 int maxLen = 0;

 while(right < baseStr.length()){
    while(hashset.count(baseStr[right]) ){
        hashset.erase(baseStr[left]);
        left+=1;
  }
    maxLen = max(maxLen , right-left+1);
    hashset.insert(baseStr[right]);
    right+=1;
 }
   
return maxLen;



}


int main(){

cout << lonSubStr("abcdfbab")<< endl;

}

File: dyswlongsubstr.py






def longSubStr(baseStr):
    right = 0
    left = 0
    hashset = set()
    maxLen = 0
    while right < len(baseStr):
        while  baseStr[right] in hashset:
             hashset.remove(baseStr[left])
             left+=1
        maxLen = max(maxLen, right - left+1)
        hashset.add(baseStr[right])
        right+=1
    return maxLen


print(longSubStr('abcdefgba'))
